File: backend/src/api/explorer/models/surface.py
import datetime
from typing import List, Optional

from fastapi_utils.api_model import APIModel
from pydantic import BaseModel


class SurfaceRequest(BaseModel):
    ticker: str
    option_type: str
    field: str
    range: int
    maturity: Optional[datetime.date] = None


class Surface(APIModel):
    x: List
    y: List
    z: List[List[float]]

File: frontend/src/models/surface.ts
export interface Surface {
    x: Date[]
    y: number[]
    z: number[][]
}

export class SurfaceRequest {
    ticker: string
    option_type: string
    field: string
    range: number
    maturity: string

    constructor(ticker: string, option_type: string, field: string, range: number, maturity: string) {
        this.field = field
        this.option_type = option_type
        this.ticker = ticker
        this.range = range
        this.maturity = maturity
    }
}